### PR TITLE
feat: member phone modal

### PR DESCRIPTION
**Pull request by LoisChen68**

### CODE REVIEW

**Comment by LoisChen68** [comment]
[(2)【名單撥打】電話能夠標記為空號](https://trello.com/c/w5ubij8L/4863-2%E3%80%90%E5%90%8D%E5%96%AE%E6%92%A5%E6%89%93%E3%80%91%E9%9B%BB%E8%A9%B1%E8%83%BD%E5%A4%A0%E6%A8%99%E8%A8%98%E7%82%BA%E7%A9%BA%E8%99%9F)

File: src/hooks/member.ts [hunk 1]
```diff
@@ -242,7 +243,12 @@ export const useMemberAdmin = (memberId: string) => {
             : null,
           tags: data.member_by_pk.member_tags.map(v => v.tag_name),
           specialities: data.member_by_pk.member_specialities.map(v => v.tag_name),
-          phones: data.member_by_pk.member_phones.map(v => v.phone).filter(v => v),
+          phones: data.member_by_pk.member_phones
+            .map(v => ({
+              isValid: v.is_valid,
+              phoneNumber: v.phone,
+            }))
+            .filter(v => v),
```
Comment (Ponchimeow) [inline src/hooks/member.ts:251]: NitPick: this filter is dispensable

File: src/components/sale/MemberPhoneModal.tsx [hunk 2]
```diff
@@ -0,0 +1,231 @@
...
+    }
+  `)
+  const [updateMemberMangerId] = useMutation(gql`
+    mutation UpdateMemberMangerId($memberId: String!) {
+      update_member(where: { id: { _eq: $memberId } }, _set: { manager_id: null }) {
+        affected_rows
+      }
+    }
+  `)
+
```
Comment (Ponchimeow) [inline src/components/sale/MemberPhoneModal.tsx:76]: updateMemberManagerId is ambiguous in the situation
1. change the naming, such as updateMemberManagerIdNull
2. passing parameters to SQL

File: src/components/sale/MemberPhoneModal.tsx [hunk 3]
```diff
@@ -0,0 +1,231 @@
...
+  const [updateMemberPhone] = useMutation(gql`
+    mutation UpdateMemberPhone($memberId: String!, $phoneNumber: String!, $isValid: Boolean!) {
+      update_member_phone(
+        where: { member_id: { _eq: $memberId }, phone: { _eq: $phoneNumber } }
+        _set: { is_valid: $isValid }
+      ) {
+        affected_rows
+      }
+    }
+  `)
```
Comment (Ponchimeow) [inline src/components/sale/MemberPhoneModal.tsx:68]: as below, could be...
1. change the naming
2. passing a parameter

File: src/hooks/member.ts (hunk 1 above)
Comment (LoisChen68) [reply src/hooks/member.ts]: adjusted

File: src/components/sale/MemberPhoneModal.tsx (hunk 3 above)
Comment (LoisChen68) [reply src/components/sale/MemberPhoneModal.tsx:68]: adjusted.

**Review by Ponchimeow — APPROVED** [review]

**Pull request merged**